src/realm/replication.cpp

```diff
@@ -62,36 +64,323 @@ Replication::version_type Replication::prepare_commit(version_type orig_version)
     return new_version;
 }
 
-void Replication::add_class(TableKey table_key, StringData, Table::Type)
+void Replication::add_class(TableKey table_key, StringData name, Table::Type type)
 {
+    if (auto logger = get_logger()) {
```

Comment:
Isn't a logger always provided so we can get rid off this check? (or can't we always provide one?)